docs/Releasing-the-plugin.md

```diff
@@ -24,24 +39,49 @@ In addition to those locations, run the `npm run since -- -r {version}` command
 
 ### Open a pull request
 
-Push your local release branch to the GitHub repository and open a pull request against `trunk`. Make sure to tag at least 2 plugin maintainers to request a review. Once the pull request is approved by at least 2 plugin maintainers, the pull request can be merged.
-
-If this is a major or minor release, please keep the release branch around since that branch can be used to later create patch releases from the same state of the codebase.
+Push your local branch to the GitHub repository and open a pull request against the release branch `release/{milestoneName}`. Make sure to tag at least 2 plugin maintainers to request a review, as usual. Once the pull request is approved by at least 2 plugin maintainers, the pull request can be merged.
 
 ## Publishing the release
 
-Once the above pull request has been merged, let the other maintainers know on [Slack](https://wordpress.slack.com/archives/performance) that no new commits or pull requests must be added to the branch due to the release process. Then, make sure that all GitHub actions successfully pass for the target branch (e.g. [for `trunk`](https://github.com/WordPress/performance/actions?query=branch%3Atrunk)).
+### Notify maintainers and review release branch
+
+Once the above pull request has been merged, let the other maintainers know on [Slack](https://wordpress.slack.com/archives/performance) that no new commits or pull requests must be added to the release branch due to the release process. Then, make sure that all [GitHub workflows](https://github.com/WordPress/performance/actions) successfully pass for the release branch.
+
+### Create the release tag
+
+After that, [create a new release tag for the plugin on GitHub](https://github.com/WordPress/performance/releases/new). The release tag should have the same name as the corresponding milestone used earlier, and it should be created from the release branch `release/{milestoneName}`. Finally, add the changelog (it can be found in the `readme.txt` file) to the release description and create the release.
+
+### Review and test the deployment
+
+Once a new version is released on GitHub, the plugin will be deployed to the [WordPress.org repository](https://wordpress.org/plugins/performance-lab/). [The deployment workflow progress can be tracked here.](https://github.com/WordPress/performance/actions/workflows/deploy-dotorg.yml)
+
+At this point, test the update process from your WordPress site, getting the plugin from wordpress.org. The main aspects to test at this point are:
+
+* Install or update the plugin via wordpress.org, ZIP upload, WP-CLI, etc.
+* Test the plugin on different WordPress core versions supported.
+* Test the plugin on a multisite environment as well.
```

Comment:
Should we add some direction with regards to local environments to test this on? Like `wp-env`?

Reply:
@dainemawer I think that would be a bit overkill here, it can be any environment where these points can be tested. It can be a local site, somebody's production site, a quickly spun up test site using one of those throwaway site providers - so IMO specifying more details here wouldn't be necessary, but if it was, it should go elsewhere since it would go too much into detail and thus make this doc too long - it's already fairly long based on the current more crucial requirements defined.

Reply:
Cool beans!